File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        maxi=nums[0]
        sum=0
        for num in nums:
            sum+=num
            maxi=max(maxi,sum)
            if sum<0:
                sum=0
        return maxi

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int n = nums.length;
        int sum=0;
        int maxi = nums[0];
        for(int v : nums){
            sum+=v;
            maxi = Math.max(maxi, sum);
            if(sum<0){
                sum=0;
            }
        }
    return maxi;       
    }
}